# Importing Libraries and Loading Data

In [102]:
import pandas as pd
df = pd.read_csv('./data/Aviation_Data.csv')
df

<ipython-input-102-1b1420092910>:2: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('./data/Aviation_Data.csv')


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2        Saltville, VA  United States  36.922223 -81.878056          NaN   
3           EUREKA, CA  United States        NaN        NaN          NaN   
4           Canton, OH  United States        NaN        NaN          NaN   
...                ...            ...        ...        ...          ...   
90343    Annapolis, MD  United States        NaN        NaN          NaN   
90344      Hampton, NH  United States        NaN        NaN          NaN   
90345       Payson, AZ  United States    341525N   1112021W          PAN   
90346       Morgan, UT  United States        NaN        NaN          NaN   
90347       Athens, GA  United States        NaN        NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               N

# Checking Data Structure, Shape and Columns

In [103]:
df.shape

(90348, 31)

In [104]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

In [105]:
df.describe(include='object')

Event.Id Investigation.Type Accident.Number  Event.Date  \
count            88889              90348           88889       88889   
unique           87951                 71           88863       14782   
top     20001212X19172           Accident      CEN22LA149  1984-06-30   
freq                 3              85015               2          25   

             Location        Country Latitude Longitude Airport.Code  \
count           88837          88663    34382     34373        50249   
unique          27758            219    25592     27156        10375   
top     ANCHORAGE, AK  United States  332739N  0112457W         NONE   
freq              434          82248       19        24         1488   

       Airport.Name  ... Amateur.Built    Engine.Type FAR.Description  \
count         52790  ...         88787          81812           32023   
unique        24871  ...             2             13              31   
top         Private  ...            No  Reciprocating             091   
freq            240  ...         80312          69530           18221   

       Schedule Purpose.of.flight Air.carrier Weather.Condition  \
count     12582             82697       16648             84397   
unique        3                26       13590                 4   
top        NSCH          Personal       Pilot               VMC   
freq       4474             49448         258             77303   

       Broad.phase.of.flight   Report.Status Publication.Date  
count                  61724           82508            73659  
unique                    12           17007             2923  
top                  Landing  Probable Cause       25-09-2020  
freq                   15428           61754            16317  

[4 rows x 26 columns]

# Cleaning Data

In [106]:
# Checking for Duplicated Values
df.duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
90343    False
90344    False
90345    False
90346    False
90347    False
Length: 90348, dtype: bool

In [107]:
# dropping the duplicates
df.drop_duplicates()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2        Saltville, VA  United States  36.922223 -81.878056          NaN   
3           EUREKA, CA  United States        NaN        NaN          NaN   
4           Canton, OH  United States        NaN        NaN          NaN   
...                ...            ...        ...        ...          ...   
90343    Annapolis, MD  United States        NaN        NaN          NaN   
90344      Hampton, NH  United States        NaN        NaN          NaN   
90345       Payson, AZ  United States    341525N   1112021W          PAN   
90346       Morgan, UT  United States        NaN        NaN          NaN   
90347       Athens, GA  United States        NaN        NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               N

In [108]:
# Changing date to date type and grouping my data

df['Event.Date'] = pd.to_datetime(df['Event.Date'], errors='coerce')
df['Event.Date'].isna().sum

<bound method NDFrame._add_numeric_operations.<locals>.sum of 0        False
1        False
2        False
3        False
4        False
         ...  
90343    False
90344    False
90345    False
90346    False
90347    False
Name: Event.Date, Length: 90348, dtype: bool>

In [109]:
# Grouping Categorical columns and numerical columns
categorical_cols = df.select_dtypes(include=['object']).columns.tolist()
numerical_cols = df.select_dtypes(include=['int64', 'float64']).columns.tolist()

In [110]:
categorical_cols

['Event.Id',
 'Investigation.Type',
 'Accident.Number',
 'Location',
 'Country',
 'Latitude',
 'Longitude',
 'Airport.Code',
 'Airport.Name',
 'Injury.Severity',
 'Aircraft.damage',
 'Aircraft.Category',
 'Registration.Number',
 'Make',
 'Model',
 'Amateur.Built',
 'Engine.Type',
 'FAR.Description',
 'Schedule',
 'Purpose.of.flight',
 'Air.carrier',
 'Weather.Condition',
 'Broad.phase.of.flight',
 'Report.Status',
 'Publication.Date']

In [111]:
numerical_cols

['Number.of.Engines',
 'Total.Fatal.Injuries',
 'Total.Serious.Injuries',
 'Total.Minor.Injuries',
 'Total.Uninjured']

# Handling Missing Data

In [112]:
# Checking for missing data

df.isna().sum()

Event.Id                   1459
Investigation.Type            0
Accident.Number            1459
Event.Date                 1459
Location                   1511
Country                    1685
Latitude                  55966
Longitude                 55975
Airport.Code              40099
Airport.Name              37558
Injury.Severity            2459
Aircraft.damage            4653
Aircraft.Category         58061
Registration.Number        2776
Make                       1522
Model                      1551
Amateur.Built              1561
Number.of.Engines          7543
Engine.Type                8536
FAR.Description           58325
Schedule                  77766
Purpose.of.flight          7651
Air.carrier               73700
Total.Fatal.Injuries      12860
Total.Serious.Injuries    13969
Total.Minor.Injuries      13392
Total.Uninjured            7371
Weather.Condition          5951
Broad.phase.of.flight     28624
Report.Status              7840
Publication.Date          16689
dtype: i

In [113]:
# replace placeholders with actual NaN
df.replace(['N/A', 'NA', 'Na', 'NONE', 'Unknown', '--', ''], pd.NA, inplace=True)

In [114]:
# dropping rows missing critical identifiers fields
df.dropna(subset=['Make','Model'], inplace=True)

In [115]:
# Replacing missing categorical data with 'unknown'
for col in categorical_cols:
    df[col].fillna('Unknown', inplace=True)

In [116]:
# fill missing no.of.engines by grouping the make and model this preserves manufacturers level patterns
df['Number.of.Engines'] = df.groupby(['Make','Model'])['Number.of.Engines'].transform(lambda x: x.fillna(x.median()))
df['Number.of.Engines'].fillna(df['Number.of.Engines'].median(), inplace=True)
df['Number.of.Engines'].isna().sum()

0

In [117]:
injuries = ['Total.Fatal.Injuries', 'Total.Serious.Injuries', 
           'Total.Minor.Injuries', 'Total.Uninjured']
for col in injuries:
    median_val = df[col].median()
    df[col].fillna(median_val, inplace=True)



In [118]:
#Standardizing Categorical Data
exclude = ['Event.Id', 'Accident.Number','Registration.Number', 'Model' ]
for col in [c for c in categorical_cols if c not in exclude]:
    df[col] = (
        df[col]
        .astype(str)                
        .str.strip()                
        .str.title()                
    )



In [119]:
df.head()

Event.Id Investigation.Type Accident.Number Event.Date  \
0  20001218X45444           Accident      SEA87LA080 1948-10-24   
1  20001218X45447           Accident      LAX94LA336 1962-07-19   
2  20061025X01555           Accident      NYC07LA005 1974-08-30   
3  20001218X45448           Accident      LAX96LA321 1977-06-19   
4  20041105X01764           Accident      CHI79FA064 1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  Moose Creek, Id  United States    Unknown     Unknown      Unknown   
1   Bridgeport, Ca  United States    Unknown     Unknown      Unknown   
2    Saltville, Va  United States  36.922223  -81.878056      Unknown   
3       Eureka, Ca  United States    Unknown     Unknown      Unknown   
4       Canton, Oh  United States    Unknown     Unknown      Unknown   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0      Unknown  ...          Personal     Unknown                  2.0   
1      Unknown  ...          Personal     Unknown                  4.0   
2      Unknown  ...          Personal     Unknown                  3.0   
3      Unknown  ...          Personal     Unknown                  2.0   
4      Unknown  ...          Personal     Unknown                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    0.0                  0.0             1.0   
3                    0.0                  0.0             0.0   
4                    2.0                  0.0             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               Unk                 Cruise  Probable Cause          Unknown  
1               Unk                Unknown  Probable Cause       19-09-1996  
2               Imc                 Cruise  Probable Cause       26-02-2007  
3               Imc                 Cruise  Probable Cause       12-09-2000  
4               Vmc               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [120]:
# Saving the cleaned dataset

df.to_csv("Clean_Aviation_Data.csv", index=False)

In [121]:
df.isna().sum()

Event.Id                  0
Investigation.Type        0
Accident.Number           0
Event.Date                0
Location                  0
Country                   0
Latitude                  0
Longitude                 0
Airport.Code              0
Airport.Name              0
Injury.Severity           0
Aircraft.damage           0
Aircraft.Category         0
Registration.Number       0
Make                      0
Model                     0
Amateur.Built             0
Number.of.Engines         0
Engine.Type               0
FAR.Description           0
Schedule                  0
Purpose.of.flight         0
Air.carrier               0
Total.Fatal.Injuries      0
Total.Serious.Injuries    0
Total.Minor.Injuries      0
Total.Uninjured           0
Weather.Condition         0
Broad.phase.of.flight     0
Report.Status             0
Publication.Date          0
dtype: int64

In [122]:
df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

# Data Analysis

Objectives:
Your company wants to start operating aircraft for commercial and private use but lacks aviation experience.
We’ll analyze accident data to identify:

1. Which aircraft are the safest (lowest risk)?

2. What factors increase or decrease accident severity?

3. What operational strategies reduce exposure to high-risk conditions?

In [123]:
import pandas as pd
import numpy as np

# Severity Index (weighted injuries)
df['Severity_Index'] = (
    df['Total.Fatal.Injuries'] * 3 +
    df['Total.Serious.Injuries'] * 2 +
    df['Total.Minor.Injuries']
)

# Any fatality flag
df['Any_Fatal'] = (df['Total.Fatal.Injuries'] > 0).astype(int)

# Convert Damage Levels to numeric for averaging
damage_map = {'Destroyed':3, 'Substantial':2, 'Minor':1, 'None':0, 'Unknown':np.nan}
df['Damage_Score'] = df['Aircraft.damage'].map(damage_map)
df['Damage_Score'] = df['Damage_Score'].fillna(df['Damage_Score'].median())


# Which Aircraft Are Lowest Risk?

In [101]:
# Aggregate risk metrics by Make and Model
model_risk = (
    df.groupby(['Make','Model'])
      .agg(
          Total_Accidents=('Event.Id','count'),
          Mean_Severity=('Severity_Index','mean'),
          Fatal_Rate=('Any_Fatal','mean'),
          Mean_Damage=('Damage_Score','mean')
      )
      .reset_index()
)

# Filter for models with enough data (≥15 accidents)
model_risk = model_risk[model_risk['Total_Accidents'] >= 15]

# Normalize and compute Composite Risk
def minmax(series):
    if series.max() == series.min():
        return pd.Series(0, index=series.index)
    return (series - series.min()) / (series.max() - series.min())

for c in ['Mean_Severity','Fatal_Rate','Mean_Damage']:
    model_risk[f'Norm_{c}'] = minmax(model_risk[c])

model_risk['Composite_Risk'] = model_risk[['Norm_Mean_Severity','Norm_Fatal_Rate','Norm_Mean_Damage']].mean(axis=1)

# Show top safest aircraft
lowest_risk = model_risk.sort_values('Composite_Risk').head(10)
lowest_risk[['Make','Model','Total_Accidents','Mean_Severity','Fatal_Rate','Mean_Damage','Composite_Risk']]
lowest_risk


Make        Model  Total_Accidents  Mean_Severity  \
11755  Mcdonnell Douglas        MD-80               16       4.000000   
3324              Boeing          747               85       2.094118   
3779      Bombardier Inc  CL-600-2B19               18       0.388889   
3493              Boeing          777               95       0.810526   
11761  Mcdonnell Douglas        MD-88               21       1.619048   
3519              Boeing          787               26       0.500000   
3304              Boeing      737-7H4               19       0.894737   
3468              Boeing      767-300               15       1.400000   
3151              Boeing      727-200               43       5.232558   
7275             Embraer    EMB-145LR               21       1.380952   

       Fatal_Rate  Mean_Damage  Norm_Mean_Severity  Norm_Fatal_Rate  \
11755    0.000000     1.562500            0.068319         0.000000   
3324     0.023529     1.576471            0.034973         0.035294   
3779     0.000000     1.666667            0.005138         0.000000   
3493     0.000000     1.673684            0.012515         0.000000   
11761    0.047619     1.571429            0.026661         0.071429   
3519     0.000000     1.692308            0.007082         0.000000   
3304     0.000000     1.736842            0.013988         0.000000   
3468     0.000000     1.733333            0.022829         0.000000   
3151     0.023256     1.627907            0.089884         0.034884   
7275     0.000000     1.761905            0.022495         0.000000   

       Norm_Mean_Damage  Composite_Risk  
11755          0.134466        0.067595  
3324           0.145287        0.071851  
3779           0.215146        0.073428  
3493           0.220582        0.077699  
11761          0.141382        0.079824  
3519           0.235006        0.080696  
3304           0.269499        0.094496  
3468           0.266781        0.096537  
3151           0.185126        0.103298  
7275           0.288911        0.103802

# Findings
Grouping accident data by Make + Model reveals that composite risk (combined average of severity, fatality rate, and damage) is lowest for newer, large-transport jets such as the Boeing 747, 787, 737-7H4 and the Bombardier CL-600-2B19 regional jet. while smaller or older models (e.g., McDonnell Douglas DC-9, MD-80) show higher severity averages.

# Recommendation 1
Begin operations with modern transport-category aircraft (e.g., Boeing 737-800 or 787) for initial fleet acquisition.
These models show the lowest historical accident severity, suggesting the best risk-to-profit ratio for a new operator.

# Weather Conditions and Accident Severity

In [124]:
# Weather Analysis
weather_summary = (
    df.groupby('Weather.Condition')
      .agg(
          Total_Accidents=('Event.Id', 'count'),
          Mean_Severity=('Severity_Index', 'mean'),
          Fatal_Rate=('Any_Fatal', 'mean'),
          Mean_Damage=('Damage_Score', 'mean')
      )
      .reset_index()
      .sort_values('Mean_Severity', ascending=False)
)

weather_summary


Weather.Condition  Total_Accidents  Mean_Severity  Fatal_Rate  Mean_Damage
1               Unk             1116       8.522401    0.509857     2.392473
2           Unknown             4439       7.673350    0.339716     2.097319
0               Imc             5971       7.113046    0.579132     2.531067
3               Vmc            77224       1.740171    0.158409     2.152220

In [125]:
weather_summary.to_csv("factor_Weather_Condition.csv", index=False)


# Findings
Accidents occurring in IMC Weather conditions are roughly four times more severe and three-to-four times more likely to be fatal than those in VMC Weather conditions. Mean Damage also increases, indicating that aircraft in IMC crashes suffer more destruction.

# Recommendation 2
Invest early in IMC-capable aircraft, IFR-rated pilots, and weather-driven dispatch policies.

# How the phase of flight and purpose of flight relate to accident severity and fatality rates.

In [126]:
# Accident severity by phase of flight
phase_summary = (
    df.groupby('Broad.phase.of.flight')
      .agg(
          Total_Accidents=('Event.Id', 'count'),
          Mean_Severity=('Severity_Index', 'mean'),
          Fatal_Rate=('Any_Fatal', 'mean'),
          Mean_Damage=('Damage_Score', 'mean')
      )
      .reset_index()
      .sort_values('Mean_Severity', ascending=False)
)

phase_summary


Broad.phase.of.flight  Total_Accidents  Mean_Severity  Fatal_Rate  \
11               Unknown            27638       3.766734    0.237716   
1                  Climb             2030       3.668966    0.303941   
0               Approach             6533       2.733048    0.243380   
6            Maneuvering             8133       2.673429    0.390754   
2                 Cruise            10259       2.670533    0.271371   
7                  Other              119       2.537815    0.378151   
3                Descent             1885       2.483289    0.242440   
4              Go-Around             1353       2.334812    0.197339   
9                Takeoff            12475       1.933467    0.145892   
8               Standing              945       1.441270    0.113228   
10                  Taxi             1958       0.525536    0.020940   
5                Landing            15422       0.468616    0.018934   

    Mean_Damage  
11     2.109487  
1      2.319212  
0      2.263126  
6      2.443748  
2      2.308120  
7      2.218487  
3      2.253581  
4      2.282336  
9      2.206493  
8      1.949206  
10     1.903473  
5      2.026456

In [127]:
# Accident Severity by Phase of flight
purpose_summary = (
    df.groupby('Purpose.of.flight')
      .agg(
          Total_Accidents=('Event.Id', 'count'),
          Mean_Severity=('Severity_Index', 'mean'),
          Fatal_Rate=('Any_Fatal', 'mean'),
          Mean_Damage=('Damage_Score', 'mean')
      )
      .reset_index()
      .sort_values('Mean_Severity', ascending=False)
)

purpose_summary


Purpose.of.flight  Total_Accidents  Mean_Severity  Fatal_Rate  \
5                        Asho                6       7.333333    0.666667   
25                    Unknown            12922       6.906748    0.240830   
24                  Skydiving              182       5.159341    0.324176   
8         Executive/Corporate              553       4.108499    0.254973   
2                    Air Drop               11       3.636364    0.272727   
11               Firefighting               40       3.300000    0.475000   
1          Aerial Observation              794       2.769521    0.280856   
4               Air Race/Show               59       2.610169    0.491525   
19            Public Aircraft              720       2.555556    0.243056   
7                    Business             4016       2.441982    0.264940   
22    Public Aircraft - State               64       2.296875    0.187500   
15             Other Work Use             1264       2.295886    0.206487   
21    Public Aircraft - Local               74       2.108108    0.121622   
10                      Ferry              812       1.967980    0.208128   
20  Public Aircraft - Federal              105       1.923810    0.190476   
16                   Personal            49390       1.828872    0.209698   
3               Air Race Show               99       1.797980    0.393939   
9               External Load              123       1.699187    0.243902   
17                Positioning             1645       1.634650    0.238298   
12                Flight Test              404       1.618812    0.205446   
13                 Glider Tow               53       1.452830    0.264151   
6                  Banner Tow              101       1.277228    0.158416   
14              Instructional            10598       1.023872    0.096245   
0          Aerial Application             4710       0.770701    0.109766   
23                       Pubs                4       0.500000    0.000000   
18                       Publ                1       0.000000    0.000000   

    Mean_Damage  
5      2.666667  
25     2.083811  
24     2.159341  
8      2.227848  
2      2.363636  
11     2.375000  
1      2.268262  
4      2.288136  
19     2.262500  
7      2.268675  
22     2.093750  
15     2.175633  
21     2.027027  
10     2.256158  
20     2.171429  
16     2.203381  
3      2.141414  
9      2.130081  
17     2.206687  
12     2.150990  
13     2.075472  
6      2.089109  
14     2.099264  
0      2.220807  
23     2.000000  
18     2.000000

In [128]:
phase_summary.to_csv("factor_Broad_phase_of_flight.csv", index=False)
purpose_summary.to_csv("factor_Purpose_of_flight.csv", index=False)


# Findings
Grouped results show that severity and fatality are highest in Maneuvering, Climb, and Cruise phases, and lowest during Landing.
By purpose, Instructional and Aerial Application flights are least severe, while Personal and Business flights show higher severity and fatal rates. Accidents during high-energy or unstructured operations are far more severe.
Structured, supervised missions reduce the risk of catastrophic outcomes.

# Recommendations 3
Implement strict Standard Operating Procedures (SOPs), limit early operations to structured mission types (charter, training), and emphasize recurrent training.
This reduces operational risk, increases reliability, and strengthens regulatory compliance for a new aviation division.